lightningd/opening_control.c

```diff
@@ -1202,6 +1203,184 @@ static struct command_result *json_fundchannel_start(struct command *cmd,
+	
+	/*To indicate this is an stub channel we keep it's scid to 1x1x1.*/
+	if(!short_channel_id_from_str("1x1x1", 6, scid)){
+		return NULL;
+	}
+	
+	struct channel *channel; //Channel Shell with Dummy data(mostly)
+
+	channel = new_channel(peer, id,
+			      NULL, /* No shachain yet */
+			      CHANNELD_NORMAL,
+			      LOCAL,
+			      peer->ld->log,
+			      "billboard",
+			      8, &our_config,
+			      101,
+			      1, 1, 1,
+			      &funding,
+			      AMOUNT_SAT(500000000), // FIXME: Is this correct? 
+				  /*Need to keep this very high from trimming the fees 
+				  	in onchaind.c(main).*/
+			      AMOUNT_MSAT(0),
+			      AMOUNT_SAT(0),
+			      true, /* !remote_funding_locked */
+			      scid, /* no scid yet */
+			      &cid,
+			      /* The three arguments below are msatoshi_to_us,
+			       * msatoshi_to_us_min, and msatoshi_to_us_max.
+			       * Because, this is a newly-funded channel,
+			       * all three are same value. */
+			      AMOUNT_MSAT(0),
+			      AMOUNT_MSAT(0), /* msat_to_us_min */
+			      AMOUNT_MSAT(0), /* msat_to_us_max */
+			      NULL,
+			      &sig,
+			      NULL, /* No HTLC sigs yet */
+			      channel_info,
+			      new_fee_states(ld->wallet, LOCAL, &feerate),
+			      NULL, /* No shutdown_scriptpubkey[REMOTE] yet */
+			      NULL,
+			      1, false,
+			      NULL, /* No commit sent yet */
+			      /* If we're fundee, could be a little before this
+			       * in theory, but it's only used for timing out. */
+			      get_network_blockheight(ld->topology),
+			      100, 10000,
+			      /* We are connected */
+			      false,
+			      &basepoints,
+			   	  &localFundingPubkey,
+			      NULL,
+			      ld->config.fee_base,
+			      ld->config.fee_per_satoshi,
+			      NULL,
+			      0, 0,
+			      channel_type_static_remotekey(NULL),
+			      NUM_SIDES, /* closer not yet known */
+			      REASON_REMOTE,
+			      NULL,
+			      take(new_height_states(ld->wallet, LOCAL,
+						    &blockht)),
+			      0, NULL, 0, 0, /* No leases on v1s */
+			      ld->config.htlc_minimum_msat,
+			      ld->config.htlc_maximum_msat);
+
+	return channel;
+}
+static struct command_result *json_commit_channel(struct command *cmd,
+						     const char *buffer,
+						     const jsmntok_t *obj UNNEEDED,
+						     const jsmntok_t *params)
+{
+	u64 id;
+	struct node_id nodeid;
+	struct channel_id cid;
+	struct bitcoin_outpoint funding;
+	struct wireaddr_internal addr;
+	const char *scb;
+
+	if (!param(cmd, buffer, params,
+		p_req("scb", param_string, &scb),
+		NULL))
+		return command_param_failed();
+	char *copy = strdup(scb);
+
+	char *token = strtok(copy, ",");
+	while(token != NULL){
+
+		u8 *scb_arr = tal_hexdata(cmd, token, strlen(token));
+		if(!fromwire_static_chan_backup(scb_arr, 
+										&id, 
+										&cid, 
+										&nodeid, 
+										&addr, 
+										&funding)){
+			/* FIXME: INSERT BROKEN STATEMENT HERE!*/
```

Comment:
If you do this inside the param parser, it's natural: return an error that the scb is invalid.